File: Tests/CSharp/OutputDirectory/OutputDirectory.cs
// **********************************************************************
//
// Copyright (c) 2009-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace Test
{
    public class OutputDirectory
    {
        public static bool outputDirectory()
        {
            String baseDir = Path.GetFullPath(".");
            while (!baseDir.EndsWith("CSharp"))
            {
                baseDir = Path.GetDirectoryName(baseDir);
            }
            return File.Exists(Path.Combine(baseDir, "OutputDirectory", "generated", "client", "Test.cs"));
        }
    }
}

File: Tests/Cpp/OutputDirectory/OutputDirectory.cpp
// **********************************************************************
//
// Copyright (c) 2009-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice\Ice.h>
#include "OutputDirectory.h"
#include <Test.h>

#include <sys/types.h>
#include <sys/stat.h>

bool Test::outputDirectory()
{
	struct _stat buf;
	return _stat("../../OutputDirectory/generated/client/Test.cpp", &buf) == 0 &&
		   _stat("../../OutputDirectory/generated/client/Test.h", &buf) == 0;
}
